pkg/proc.attr.hydfab/R/proc_attr_grabber.R

```diff
@@ -15,6 +15,90 @@ library(data.table)
+
+  datasets <- base::unlist(raw_config$formulation_metadata)[['datasets']]
+  ds_type <- try(base::unlist(raw_config$file_io)[['ds_type']])
+  if('try-error' %in% base::class(ds_type) || is.null(ds_type)){
+    warning('ds_type undefined in the attribute config file. It is generally
+    expected to be "training" or "prediction"')
+    ds_type <- '' # !!! Generally expected to be 'training' or 'prediction' !!!
+  }
+  write_type <- try(base::unlist(raw_config$file_io[['write_type']]))
+  if('try-error' %in% base::class(write_type) || is.null(write_type)){
+    write_type <- 'parquet'
+  }
+
+  # Figure out the dataset name(s) in order to generate path_meta appropriately
+  path_meta <- base::unlist(raw_config$file_io)[['path_meta']] # Still needs glue substitution
+
+
+  # Read s3 connection details
+  s3_base <- base::unlist(raw_config$hydfab_config)[['s3_base']]#s3://lynker-spatial/tabular-resources" # s3 path containing hydrofabric-formatted attribute datasets
+  s3_bucket <- base::unlist(raw_config$hydfab_config)[['s3_bucket']] #'lynker-spatial' # s3 bucket containing hydrofabric data
+
+  # s3 path to hydroatlas data formatted for hydrofabric
+  if ("s3_path_hydatl" %in% names(base::unlist(raw_config$attr_select))){
+    s3_path_hydatl <- glue::glue(base::unlist(raw_config$attr_select)[['s3_path_hydatl']])  # glue::glue('{s3_base}/hydroATLAS/hydroatlas_vars.parquet')
+  } else {
+    s3_path_hydatl <- NULL
+  }
+
+  # Additional config options
+  hf_cat_sel <-  base::unlist(raw_config$hydfab_config)[['hf_cat_sel']] #c("total","all")[1] # total: interested in the single location's aggregated catchment data; all: all subcatchments of interest
+  ext <- base::unlist(raw_config$hydfab_config)[['ext']] # 'gpkg'
+
+  #-----------------------------------------------------
+  # Variable listings:
+  names_attr_sel <- base::unlist(base::lapply(raw_config$attr_select,
+                                              function(x) base::names(x)))
+
+  # Transform into single named list of lists rather than nested sublists
+  idxs_vars <- base::grep("_vars", names_attr_sel)
+  var_names <- names_attr_sel[idxs_vars]
+  sub_attr_sel <- base::lapply(idxs_vars, function(i)
+    raw_config$attr_select[[i]][[1]])
+  base::names(sub_attr_sel) <- var_names
+
+  # Subset to only those non-null variables:
+  sub_attr_sel <- sub_attr_sel[base::unlist(base::lapply(sub_attr_sel,
+                                                         function(x) base::any(!base::is.null(unlist(x)))))]
+  var_names_sub <- names(sub_attr_sel)
+  #-----------------------------------------------------
+
+  Retr_Params <- base::list(paths = base::list(
+    # Note that if a path is provided, ensure the
+    # name includes 'path'. Same for directory having variable name with 'dir'
+    dir_db_hydfab=dir_db_hydfab,
```

Comment:
consistency in formatting ... name = value, or name=value. I think most R standards suggest spaces around "=".